src/GafferScene/SetAlgo.cpp

```diff
@@ -449,6 +449,22 @@ IECore::MurmurHash setExpressionHash( const std::string &setExpression, const Sc
 	return h;
 }
 
+bool affectsSetExpression( const Plug *scenePlugChild )
+{
+	const ScenePlug *parent = scenePlugChild->parent<ScenePlug>();
+	if( !parent || parent->direction() != Gaffer::Plug::In )
+	{
+		return false;
+	}
+
+	if( parent->attributesPlug() == scenePlugChild || parent->setPlug() == scenePlugChild )
```

Comment:
Does the attributesPlug really affect the result of a set expression? I think this is mixing the implementation of the EvaluateLightLinks node with the more general-purpose set expression code.